fix: Unlink and delete batch created from stock reco on cancel

marination opened the pull request:
Port of https://github.com/frappe/erpnext/pull/23192/

marination committed:
fix: Unlink and delete batch created from stock reco on cancel

marination committed:
chore: Tests for Stock Entry

marination committed:
chore: Tests for Stock Reconciliation

marination committed:
fix: Item Alternative Test

ankush commented on erpnext/stock/doctype/stock_entry/test_stock_entry.py:
```diff
@@ -653,6 +653,109 @@ def test_serial_cancel(self):
+
+		batch_qty = get_batch_qty(batch_no, "_Test Warehouse - _TC", item.item_code)
+		self.assertEqual(batch_qty, 2)
+		frappe.db.commit()
```
Commit/rollbacks are probably not required. savepoints can be added if reqd 😅

marination replied:
Yup fixing these locally

marination committed:
fix: Wrap SLE actual_qty in `flt` to avoid NoneType operation

- Since Batch cancellation SLEs do not set qtys (will fix separately), `merge_similar_entries` gets `actual_qty` as None
- This causes NoneType operation error on tests that cancel batch-serial reco
- Modified tests to avoid using commit and rollback explicitly

marination committed:
fix: Remove commit from stock entry test. The assertion is not important

marination committed:
style: Spaces to Tabs
